services/spar/src/Spar/API.hs

```diff
@@ -248,25 +250,49 @@ idpDelete zusr idpid = withDebugLog "idpDelete" (const Nothing) $ do
     Data.deleteIdPConfig idpid issuer team
     Data.deleteIdPRawMetadata idpid
   return NoContent
+  where
+    updateOldIssuers :: IdP -> Spar ()
+    updateOldIssuers _ = pure ()
+    -- we *could* update @idp ^. SAML.idpExtraInfo . wiReplacedBy@ to not keep the idp about
+    -- to be deleted in its old issuers list, but it's tricky to avoid race conditions, and
+    -- there is little to be gained here: we only use old issuers to find users that have not
+    -- been migrated yet, and if an old user points to a deleted idp, it just means that we
+    -- won't find any users to migrate.  still, doesn't hurt mucht to look either.  so we
+    -- leave old issuers dangling for now.
```

Comment:
Instead of dangling, could there be an extra field added which would be set here, "marking" these somehow, for a future cleanup?

Reply:
you can easily do the future cleanup by checking for each issuer if there are users *or* the idp entry for it.  if neither, they can be safely removed.